compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Utils/AMDAIELogicalObjFifoSplittingUtils.cpp

```diff
@@ -547,18 +567,9 @@ FailureOr<OffsetConfig> getNewOffsetConfig(
     ArrayRef<OpFoldResult> strides, size_t sizeAfterSplit, int64_t splitDimSize,
     int64_t splitStride, int64_t splitFactor,
     function_ref<InFlightDiagnostic()> emitError) {
-  SmallVector<size_t> offsetIndices;
-  for (auto iter : llvm::enumerate(llvm::zip(strides, offsets))) {
-    std::optional<int64_t> maybeOffset =
-        getConstantIntValue(std::get<1>(iter.value()));
-    bool strideEqualToSizeAfterSplit =
-        isConstantIntValue(std::get<0>(iter.value()), sizeAfterSplit);
-    bool dynamicOrNonZeroOffset =
-        !maybeOffset.has_value() || maybeOffset.value() != 0;
-    if (strideEqualToSizeAfterSplit && dynamicOrNonZeroOffset) {
-      offsetIndices.push_back(iter.index());
-    }
-  }
+  SmallVector<size_t> offsetIndices =
+      getStrideIndicesWithDynamicOrNonZeroOffset(offsets, strides,
+                                                 sizeAfterSplit);
   if (offsetIndices.size() > 1)
```

Comment:
Here also need to add a condition to check if `offsetIndices` is empty.

Reply:
I added it and simplified the remainder (`offsetIndices.size() == 1` case)